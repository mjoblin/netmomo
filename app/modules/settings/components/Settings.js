import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Input, InputNumber, Button } from 'antd';

import ConnectionStatus from 'AppRoot/components/ConnectionStatus';
import appModule from 'AppRoot/modules/app';

import * as actionCreators from '../actions.js';
import { getSettings } from '../selectors';
import './style.scss';


const Settings = ({ settings, actions, shiftyConnectionStatus }) => {
    const shiftyConnected = appModule.constants.SHIFTY_CONNECTED;
    const shiftyDisconnected = appModule.constants.SHIFTY_DISCONNECTED;
    const shiftyReconnecting = appModule.constants.SHIFTY_RECONNECTING;

    let shiftyActionButtonText, shiftyActionButtonClickHandler;

    if (shiftyConnectionStatus === shiftyConnected) {
        shiftyActionButtonText = 'Disconnect';
        shiftyActionButtonClickHandler = actions.shiftyDisconnect;
    } else if (shiftyConnectionStatus === shiftyReconnecting) {
        shiftyActionButtonText = 'Cancel Reconnect';
        shiftyActionButtonClickHandler = actions.shiftyCancelReconnect;
    } else  {
        shiftyActionButtonText = 'Connect';
        shiftyActionButtonClickHandler =
            () => { actions.shiftyConnect(settings.shiftyHost, settings.shiftyPort); };
    }

    return (
        <div className="settings">
            <h2>Settings</h2>

            <div>
                <span className="label">Shifty location:</span>
                <Input
                    className="value value-shifty-host"
                    defaultValue={settings.shiftyHost}
                    onChange={e => actions.setShiftyHost(e.target.value)}
                />

                <span className="label">Port:</span>
                <InputNumber
                    className="value"
                    defaultValue={11348}
                    min={1}
                    max={65535}
                    onChange={e => typeof(e) === 'number' && actions.setShiftyPort(e)}
                />
                <Button
                    type="primary"
                    onClick={shiftyActionButtonClickHandler}
                    disabled={
                        shiftyConnectionStatus !== shiftyConnected &&
                        shiftyConnectionStatus !== shiftyDisconnected &&
                        shiftyConnectionStatus !== shiftyReconnecting
                    }
                >
                    {shiftyActionButtonText}
                </Button>
            </div>

            <div>
                <span className="label">Shifty connection status:</span>
                <ConnectionStatus className="value" />
            </div>
        </div>
    );
};

Settings.propTypes = {
    actions: PropTypes.object.isRequired,
    settings: PropTypes.object.isRequired,
    shiftyConnectionStatus: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
    settings: getSettings(state),
    shiftyConnectionStatus: appModule.selectors.shiftyConnectionStatus(state),
});

const mapDispatchToProps = dispatch => {
    return {
        actions: bindActionCreators({
            ...actionCreators,
            shiftyConnect: appModule.actions.shiftyConnect,
            shiftyDisconnect: appModule.actions.shiftyDisconnect,
            shiftyCancelReconnect: appModule.actions.shiftyCancelReconnect,
        }, dispatch)
    };
};


export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Settings);