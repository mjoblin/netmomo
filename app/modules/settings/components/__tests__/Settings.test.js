import React from 'react';
import { shallow } from 'enzyme';
import { Settings } from '../Settings';

const testSettings = {
    shiftyHost: 'localhost',
    shiftyPort: 11348,
};

const testActions = {
    setShiftyHost: () => {},
    setShiftyPort: () => {},
    shiftyConnect: () => {},
    shiftyDisconnect: () => {},
    shiftyCancelReconnect: () => {},
};

describe('Settings', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <Settings
                actions={testActions}
                settings={testSettings}
                shiftyConnectionStatus={'Connected'}
            />
        );

        expect(component).toMatchSnapshot();
    });
});