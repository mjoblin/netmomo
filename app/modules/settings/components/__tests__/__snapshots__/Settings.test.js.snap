// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings
    actions={
      Object {
        "setShiftyHost": [Function],
        "setShiftyPort": [Function],
        "shiftyCancelReconnect": [Function],
        "shiftyConnect": [Function],
        "shiftyDisconnect": [Function],
      }
    }
    settings={
      Object {
        "shiftyHost": "localhost",
        "shiftyPort": 11348,
      }
    }
    shiftyConnectionStatus="Connected"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Settings
        </h2>,
        <div>
          <span
            className="label"
          >
            Dumpling hub host:
          </span>
          <Input
            autosize={false}
            className="value value-shifty-host"
            defaultValue="localhost"
            disabled={false}
            onChange={[Function]}
            prefixCls="ant-input"
            type="text"
          />
          <span
            className="label"
          >
            port:
          </span>
          <InputNumber
            className="value"
            defaultValue={11348}
            max={65535}
            min={1}
            onChange={[Function]}
            prefixCls="ant-input-number"
            step={1}
          />
          <Button
            clicked={false}
            disabled={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
            type="primary"
          >
            Disconnect
          </Button>
        </div>,
        <div>
          <span
            className="label"
          >
            Dumpling hub connection status:
          </span>
          <Connect(ConnectionStatus)
            className="value"
          />
        </div>,
      ],
      "className": "settings",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Settings",
        },
        "ref": null,
        "rendered": "Settings",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="label"
            >
              Dumpling hub host:
            </span>,
            <Input
              autosize={false}
              className="value value-shifty-host"
              defaultValue="localhost"
              disabled={false}
              onChange={[Function]}
              prefixCls="ant-input"
              type="text"
            />,
            <span
              className="label"
            >
              port:
            </span>,
            <InputNumber
              className="value"
              defaultValue={11348}
              max={65535}
              min={1}
              onChange={[Function]}
              prefixCls="ant-input-number"
              step={1}
            />,
            <Button
              clicked={false}
              disabled={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
              type="primary"
            >
              Disconnect
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Dumpling hub host:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Dumpling hub host:",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "autosize": false,
              "className": "value value-shifty-host",
              "defaultValue": "localhost",
              "disabled": false,
              "onChange": [Function],
              "prefixCls": "ant-input",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "port:",
              "className": "label",
            },
            "ref": null,
            "rendered": "port:",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "className": "value",
              "defaultValue": 11348,
              "max": 65535,
              "min": 1,
              "onChange": [Function],
              "prefixCls": "ant-input-number",
              "step": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Disconnect",
              "clicked": false,
              "disabled": false,
              "ghost": false,
              "loading": false,
              "onClick": [Function],
              "prefixCls": "ant-btn",
              "type": "primary",
            },
            "ref": null,
            "rendered": "Disconnect",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="label"
            >
              Dumpling hub connection status:
            </span>,
            <Connect(ConnectionStatus)
              className="value"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Dumpling hub connection status:",
              "className": "label",
            },
            "ref": null,
            "rendered": "Dumpling hub connection status:",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "className": "value",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Settings
          </h2>,
          <div>
            <span
              className="label"
            >
              Dumpling hub host:
            </span>
            <Input
              autosize={false}
              className="value value-shifty-host"
              defaultValue="localhost"
              disabled={false}
              onChange={[Function]}
              prefixCls="ant-input"
              type="text"
            />
            <span
              className="label"
            >
              port:
            </span>
            <InputNumber
              className="value"
              defaultValue={11348}
              max={65535}
              min={1}
              onChange={[Function]}
              prefixCls="ant-input-number"
              step={1}
            />
            <Button
              clicked={false}
              disabled={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
              type="primary"
            >
              Disconnect
            </Button>
          </div>,
          <div>
            <span
              className="label"
            >
              Dumpling hub connection status:
            </span>
            <Connect(ConnectionStatus)
              className="value"
            />
          </div>,
        ],
        "className": "settings",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Settings",
          },
          "ref": null,
          "rendered": "Settings",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="label"
              >
                Dumpling hub host:
              </span>,
              <Input
                autosize={false}
                className="value value-shifty-host"
                defaultValue="localhost"
                disabled={false}
                onChange={[Function]}
                prefixCls="ant-input"
                type="text"
              />,
              <span
                className="label"
              >
                port:
              </span>,
              <InputNumber
                className="value"
                defaultValue={11348}
                max={65535}
                min={1}
                onChange={[Function]}
                prefixCls="ant-input-number"
                step={1}
              />,
              <Button
                clicked={false}
                disabled={false}
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
                type="primary"
              >
                Disconnect
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Dumpling hub host:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Dumpling hub host:",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "autosize": false,
                "className": "value value-shifty-host",
                "defaultValue": "localhost",
                "disabled": false,
                "onChange": [Function],
                "prefixCls": "ant-input",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "port:",
                "className": "label",
              },
              "ref": null,
              "rendered": "port:",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "className": "value",
                "defaultValue": 11348,
                "max": 65535,
                "min": 1,
                "onChange": [Function],
                "prefixCls": "ant-input-number",
                "step": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Disconnect",
                "clicked": false,
                "disabled": false,
                "ghost": false,
                "loading": false,
                "onClick": [Function],
                "prefixCls": "ant-btn",
                "type": "primary",
              },
              "ref": null,
              "rendered": "Disconnect",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="label"
              >
                Dumpling hub connection status:
              </span>,
              <Connect(ConnectionStatus)
                className="value"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Dumpling hub connection status:",
                "className": "label",
              },
              "ref": null,
              "rendered": "Dumpling hub connection status:",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "className": "value",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
